.github/workflows/ci.yml

```diff
@@ -0,0 +1,62 @@
+name: CI
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+
+  unit-test:
+    runs-on: ubuntu-22.04
+    container:
+      image: heroku/heroku:${{ matrix.stack_number }}-build
+    strategy:
+      matrix:
+        stack_number: ["18", "20", "22"]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Unit tests on heroku:${{ matrix.stack_number }}-build
+        run: test/unit
+
+  functional-test:
+    runs-on: ubuntu-22.04
+    container:
+      image: heroku/heroku:${{ matrix.stack_number }}-build
+    strategy:
+      matrix:
+        stack_number: ["18", "20", "22"]
+    env:
+      STACK: heroku-${{ matrix.stack_number }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Functional tests on heroku:${{ matrix.stack_number }}-build
+        run: test/run
+
+  hatchet-test:
+    runs-on: ubuntu-22.04
+    container:
+      image: ruby:2.7.2
```

Comment:
Using setup-ruby instead of a subcontainer will be faster, since it will use proper caching:
https://github.com/heroku/heroku-buildpack-python/blob/05288ebd7f54d350175024e7dba396e7995c43a5/.github/workflows/ci.yml#L48-L52

The setup-ruby action also does the bundle install, so that step can then be removed from `etc/ci-setup.sh`.

Reply:
Nice. Will do. `etc/ci-setup.sh` is used in nodebin too, but it's already using setup-ruby.